### DISCUSSION TITLE
Base image with VS 2017 build tools

### DISCUSSION THREAD

**Discussion by Alvin-Prasla** [dotnet/dotnet-docker#3322] [Q&A] [answered]
Any help is appreciated. I am looking for a base image that includes VS 2017 build tools. I am trying to create a dockerfile to support **.NET 4.6.1 framework** and compile using **VS 2017 build tools**.

### Below code works: (VS 2015 build tools)
```
# escape=`
FROM microsoft/dotnet-framework:4.7.2-sdk-windowsservercore-ltsc2016
SHELL ["powershell", "-Command", "$ErrorActionPreference = 'Stop'; $ProgressPreference = 'SilentlyContinue';"]

# Download and Install
RUN write-host "*********** Download and Install Microsoft Build Tools 2015"
ADD https://download.microsoft.com/download/E/E/D/EEDF18A8-4AED-4CE0-BEBE-70A83094FC5A/BuildTools_Full.exe C:\TEMP\BuildTools_Full.exe
RUN powershell.exe Start-Process -Wait -ArgumentList "/silent" -PassThru -NoNewWindow -FilePath 'C:\TEMP\BuildTools_Full.exe'
```

### Below code failes: (VS 2017 build tools)
```
# escape=`
FROM microsoft/dotnet-framework:4.7.2-sdk-windowsservercore-ltsc2016
SHELL ["powershell", "-Command", "$ErrorActionPreference = 'Stop'; $ProgressPreference = 'SilentlyContinue';"]

# Download and Install
RUN write-host "*********** Download and Install Microsoft Build Tools 2017"
ADD https://download.visualstudio.microsoft.com/download/pr/0e24e0bf-6d05-4ce0-915f-b2dd398a2e2c/ce257ba71886950324a07344293a49ba29dc4bd15fe3ac56239e9200ca3013ab/vs_BuildTools.exe C:\TEMP\vs_buildtools.exe
RUN powershell.exe Start-Process -Wait -ArgumentList "/silent" -PassThru -NoNewWindow -FilePath 'C:\TEMP\vs_buildtools.exe'
```

The only difference is the download link. Looks like something is weird about the VS 2017 download link. How to I install VS 2017 build tools or are there any existing image I can use? Thanks

Host Environment:

- Edition: Windows server 2016 Standard
- Version: 1607
- OS Build: 10.0.14393

**Comment by mthalman** [comment, marked as answer]
There is documentation for how to install VS Build Tools for each of the supported VS versions. The page for VS 2017 can be found at https://docs.microsoft.com/visualstudio/install/build-tools-container?view=vs-2017.

In your scenario, it would look something like this:

```Dockerfile
# escape=`

FROM microsoft/dotnet-framework:4.7.2-sdk-windowsservercore-ltsc2016

# Restore the default Windows shell for correct batch processing.
SHELL ["cmd", "/S", "/C"

RUN `
    # Download the Build Tools bootstrapper.
    curl -SL --output vs_buildtools.exe https://aka.ms/vs/15/release/vs_buildtools.exe `
    `
    && (start /w vs_buildtools.exe --quiet --wait --norestart --nocache `
        --installPath C:\BuildTools
        || IF "%ERRORLEVEL%"=="3010" EXIT 0) `
    `
    # Cleanup
    && del /q vs_buildtools.exe
```

Note that there is a separate for repo for the .NET Framework container images: https://github.com/microsoft/dotnet-framework-docker. This discussion post would be better suited there.

**Reply by Alvin-Prasla** [reply]
Awesome. I'll give that a shot. In future I'll post on that discussion. Thanks for letting me know.
